docs/spec.md

```diff
@@ -1709,7 +1819,7 @@ Performs element-wise comparison of `lhs` and `rhs` tensors according to
 The values of `comparison_direction` and `compare_type` have the following
 semantics:
 
-For boolean and integer element types:
+For boolean, integer or quantized element types:
```

Comment:
I would propose that we don't introduce a special case for `compare` (similarly to the proposal about integer-only ops) and use the same `dequantize_op_quantize` mechanism for it as for other elementwise ops. 

This way, we'll be able to consistently reason about all elementwise ops, and if someone wants to compare storage types, they can bitcast convert to the storage type and perform the comparison on the result.

Reply:
Makes sense. While doing so we might need to consider a few things: 

1. After the dequantize, the floating-point  operands need to be compared, but  which compare_type is use: FLOAT or TOTAL_ORDER? We can use `FLOAT` unless we have a special usecase to use TOTAL_ORDER. 

2. After applying op to the floating-point operands we will get tensor with boolean type. For quantization, one option is to convert the boolean type tensor to floating-point tensor and then do the quantization. But `dequantize_op_quantize` may not be used out of the box. 

3. Should we constraint  the  result quantized type to have the default quantization parameter (scale=1 and zero_point=0 ) and storage_type = `i1`.